src/zenml/model/model_config.py

```diff
@@ -30,24 +38,187 @@
-        default=None
-    )
+    model: Optional[Any] = None
+    model_version: Optional[Any] = None
+    user_not_yet_warned: bool = True
+
+    def __init__(
+        self,
+        name: str,
+        license: Optional[str] = None,
+        description: Optional[str] = None,
+        audience: Optional[str] = None,
+        use_cases: Optional[str] = None,
+        limitations: Optional[str] = None,
+        trade_offs: Optional[str] = None,
+        ethics: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        version: Optional[Union[ModelStages, int, str]] = None,
+        version_description: Optional[str] = None,
+        create_new_model_version: bool = False,
+        save_models_to_registry: bool = True,
+        delete_new_version_on_failure: bool = True,
+        **kwargs: Dict[str, Any],
+    ):
+        """ModelConfig class to pass into pipeline or step to set it into a model context.
+
+        Args:
+            name: The name of the model.
+            license: The license under which the model is created.
+            description: The description of the model.
+            audience: The target audience of the model.
+            use_cases: The use cases of the model.
+            limitations: The known limitations of the model.
+            trade_offs: The tradeoffs of the model.
+            ethics: The ethical implications of the model.
+            tags: Tags associated with the model.
+            version: The model version name, number or stage is optional and points model context
+                to a specific version/stage. If skipped and `create_new_model_version` is False -
+                latest model version will be used.
+            version_description: The description of the model version.
+            create_new_model_version: Whether to create a new model version during execution
+            save_models_to_registry: Whether to save all ModelArtifacts to Model Registry,
+                if available in active stack.
+            delete_new_version_on_failure: Whether to delete failed runs with new versions for later recovery from it.
+            kwargs: Other arguments.
+        """
+        super().__init__(
+            name=name,
+            license=license,
+            description=description,
+            audience=audience,
+            use_cases=use_cases,
+            limitations=limitations,
+            trade_offs=trade_offs,
+            ethics=ethics,
+            tags=tags,
+            version=version,
+            version_description=version_description,
+            create_new_model_version=create_new_model_version,
+            save_models_to_registry=save_models_to_registry,
+            delete_new_version_on_failure=delete_new_version_on_failure,
+            model=kwargs.get("model", None),
+            model_version=kwargs.get("model_version", None),
+            user_not_yet_warned=kwargs.get("user_not_yet_warned", True),
```

Comment:
Usually the cleaner way to do this in Pydantic is to have validators for each of these arguments instead of overriding the init

Reply:
Just saw you already have a root validator below, why don't you do this in the root validator instead? Then you don't need these 60LoC of init

Reply:
Init is here for a reason: by this, I control which fields are user-facing. `model, model_version, user_not_yet_warned` are not user-facing and not expected to be used by users.
Why not `PrivateAttr`? They don't go into `dict()` while we store it in DB and the setup from DB is reused in many places (e.g. each step do it), so I want to keep track, if this object was instantiated from code (has `user_not_yet_warned=True` by skipping it from user) or from DB (has `user_not_yet_warned=False` coming from DB)

Reply:
I'll redesign this

Reply:
Init is out along with overall redesign of this part.